eplaunch/workflows/base.py

```diff
@@ -0,0 +1,17 @@
+class BaseWorkflow(object):
+
+    def name(self):
+        raise NotImplementedError("name function needs to be implemented in derived workflow class")
+
+    def description(self):
+        raise NotImplementedError("description function needs to be implemented in derived workflow class")
+
+    def get_file_types(self):
+        raise NotImplementedError("get_file_types needs to be implemented in derived workflow class")
+
+    def get_extra_data(self):
+        """
+        Allows a dictionary of extra data to be generated, defaults to empty so it is not required
+        :return: Dictionary of string, string
+        """
+        return {}
```

Comment:
This class is currently the workflow base class.  Most important are the name and get_file_types functions which are used to generate the workflow list on the UI.  Ultimately this class will need more operational functions.

Reply:
Good start!